src/robusta/core/sinks/robusta/rrm/prometheus_alert_resource_manager.py

```diff
@@ -0,0 +1,208 @@
+        self.__max_allowed_rules_per_crd = MAX_ALLOWED_RULES_PER_CRD_ALERT
+        self.__label_selector_key = "release.app"
+        self.__label_selector_value = "robusta-resource-management"
+        self.init_resources_max_attempts = 3
+        self.__label_selector = f"{self.__label_selector_key}={self.__label_selector_value}"
+        self.__plural = "prometheusrules"
+        self.__group = "monitoring.coreos.com"
+        self.__group_name = "kubernetes-apps"
+        self.__version = "v1"
+        self.__k8_apiVersion = f"{self.__group}/{self.__version}"
+        self.__kind = "PrometheusRule"
+        self.__crd_name = "robusta-prometheus.rules"
+        self.__installation_namespace = INSTALLATION_NAMESPACE
+        self.__k8_api = client.CustomObjectsApi()
+        self.__alerts_config_supabase_cache: Dict[str, PrometheusAlertRule] = {}
+
+    def __in_cluster(self, clusters_target_set: List[str] = []) -> bool:
+        return "*" in clusters_target_set or self.cluster in clusters_target_set
+
+    def __list_rules_objects(self) -> Optional[Dict[str, dict]]:
+        try:
+            return self.__k8_api.list_namespaced_custom_object(
+                group=self.__group,
+                version=self.__version,
+                plural=self.__plural,
+                namespace=self.__installation_namespace,
+                label_selector=f"{self.__label_selector}"
+            )
+        except client.ApiException as e:
+            if e.status == 404:
+                return None
+            else:
+                raise e
+
+    def __exisiting_rules_objects_map(self) -> Optional[Dict[str, dict]]:
+        objs = self.__list_rules_objects()
+
+        if not objs:
+            return None
+
+        obj_items = objs.get("items", [])
+
+        result: Dict[str, dict] = {}
+        for item in obj_items:
+            name = item.get("metadata").get("name")
+            result[name] = item
+
+        return result
+
+    def __get_snapshot_body(self, name: str, rules: List[dict]):
+        return {
+            "apiVersion": self.__k8_apiVersion,
+            "kind": self.__kind,
+            "metadata": {
+                "name": name,
+                "labels": {
+                    "release": RELEASE_NAME,
+                    "role": "alert-rules",
+                    self.__label_selector_key: self.__label_selector_value
+                },
+            },
+            "spec": {
+                "groups": [
+                    {
+                        "name": self.__group_name,
+                        "rules": rules
+                    }
+                ]
+            }
+        }
+
+    def __create_cr_rules(self, name: str, active_rules: List[PrometheusAlertRule], existing_cr_obj: Optional[dict]) -> \
+            Optional[dict]:
+        try:
+            rules = [rule.to_dict() for rule in active_rules]
+            if existing_cr_obj:
+                if not existing_cr_obj.get("spec"):
```

Comment:
We cannot do this because to replace an existing_cr_obj with a newer one we need it's current properties. Failing to do so will cause k8s update errors.